src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -272,111 +305,157 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
+                                            let barrier_epoch = barrier.epoch;
+                                            yield Message::Barrier(barrier);
+                                            write_state_post_write_barrier
+                                                .post_yield_barrier(update_vnode_bitmap.clone())
+                                                .await?;
+                                            if let Some(vnode_bitmap) = update_vnode_bitmap {
+                                                // Apply Vnode Update
+                                                read_state.update_vnode_bitmap(vnode_bitmap);
+                                                initial_write_epoch = barrier_epoch;
+                                                input = stream;
+                                                initial_write_state = write_state;
+                                                continue 'recreate_consume_stream;
+                                            } else {
+                                                write_future_state =
+                                                    WriteFuture::receive_from_upstream(
+                                                        stream,
+                                                        write_state,
+                                                    );
+                                            }
                                         }
                                     }
                                     Message::Chunk(chunk) => {
                                         let start_seq_id = seq_id;
-                                        seq_id += chunk.cardinality() as SeqIdType;
-                                        let end_seq_id = seq_id - 1;
+                                        let new_seq_id = seq_id + chunk.cardinality() as SeqIdType;
+                                        let end_seq_id = new_seq_id - 1;
                                         let epoch = write_state.epoch().curr;
                                         if let Some(chunk_to_flush) = buffer.add_or_flush_chunk(
                                             start_seq_id,
                                             end_seq_id,
                                             chunk,
                                             epoch,
                                         ) {
-                                            write_future = WriteFuture::flush_chunk(
-                                                stream,
-                                                write_state,
-                                                chunk_to_flush,
-                                                epoch,
-                                                start_seq_id,
-                                                end_seq_id,
-                                            );
+                                            if clean_state && buffer.no_flushed_items() {
```

Comment:
Seems no need to check `no_flushed_item` here, because we are sure that we may be going to flush a chunk.

Reply:
Fixed, good catch.